### PR TITLE
Keep column description if meta.osmosis_keep_description is True

### PR DESCRIPTION
**Pull request by syou6162**
## Background
The `--force-inheritance` option is useful for updating columns in the propagator when the description of a column on the data lake side is updated. In the long-term operation of dbt-osmosis, you will often use the `--force-inheritance` option, but sometimes you may not want the column description to be updated with this option.

### Example
Suppose you have a column named `shop_id` in your data lake with a column description of "ID of shop". Suppose you create a data mart about a user by referencing `shop_id` in this data lake. Suppose this data mart contains a column called `shop_id`, but you want to include a description for the column "ID of the latest shop where the user made a purchase". In this case, the `--force-inheritance` option will overwrite "ID of the latest shop where the user made a purchase" with just "ID of shop" 😭 .

Of course, you should always try to name the columns better in the data mart (e.g., `latest_shop_id`), but due to historical reasons, it may sometimes be difficult to rename columns.

## What I Changed
Don't update the column description if the column's `meta.osmosis_keep_description` is True, even if the `--force-inheritance` option is used. See testing for more detailed specification.

I've modified the behavior so that the column description won't be updated if the column's `meta.osmosis_keep_description` is set to True, even when using the `--force-inheritance option`. Please refer to the tests for more detailed specifications.

### CODE REVIEW

File: tests/test_column_level_knowledge_propagator.py [hunk 1]
```diff
@@ -161,12 +172,22 @@ def test_update_undocumented_columns_with_prior_knowledge():
...
+    )
 
     assert target_node.columns["customer_id"].description == "THIS COLUMN IS UPDATED FOR TESTING"
-    assert target_node.columns["customer_id"].meta == {"my_key": "my_value"}
-    assert set(target_node.columns["customer_id"].tags) == set(["my_tag1", "my_tag2"])
+    assert target_node.columns["customer_id"].meta == {
+        "my_key": "my_value",
+        "my_new_key": "my_new_value",
+    }
+    assert set(target_node.columns["customer_id"].tags) == set(
```
Comment (syou6162) [inline tests/test_column_level_knowledge_propagator.py:188]: Changed existing behavior: `tags` and `meta` for progenitor were taken into account, but when dbt-osmosis was run, `tags` and `meta` assigned to the target node itself disappeared.

**Comment by syou6162** [comment]
@z3z1ma Could you review this pull request?

File: src/dbt_osmosis/core/column_level_knowledge_propagator.py [hunk 2]
```diff
@@ -88,6 +89,50 @@ def get_node_columns_with_inherited_knowledge(
...
+        return original_knowledge
+
+    @staticmethod
+    def _merge_prior_knowledge_with_original_knowledge(
+        prior_knowledge: ColumnLevelKnowledge,
+        original_knowledge: ColumnLevelKnowledge,
+        add_progenitor_to_meta: bool,
+        progenitor: str,
+    ) -> None:
+        if "tags" in prior_knowledge:
```
Comment (syou6162) [inline src/dbt_osmosis/core/column_level_knowledge_propagator.py:112]: Asmost same implementation with https://github.com/z3z1ma/dbt-osmosis/blob/main/src/dbt_osmosis/core/column_level_knowledge_propagator.py#L56-L74 

See also https://github.com/z3z1ma/dbt-osmosis/pull/112/files#r1342098366

**Comment by z3z1ma** [comment]
Will review this today, sorry for delay. On brief vacation. The premise makes total sense to me.

**Comment by syou6162** [comment]
@z3z1ma Please feel free to comment if you are unclear about the intent of any of the changes.

**Comment by z3z1ma** [comment]
Thanks for patience. This looks good. The test case too! 🙏 🏆 

Very nice addition. Generally I think that a column of the same name should be defined as high up in the tree as possible and it should not change without a change of column name as well, which is therefore essentially a new node. Following this pattern makes dbt osmosis work at its DRYest and most effective. But in the long tail of ongoing usage and maintenance I see what you are saying for sure. 

Using force inheritance is too aggressive if a downstream child has a proper description. So this is a happy middle ground I think. We could consider a flag that accepts. comma separated list of column_names to exclude from propagation which gives you intra-model control but I think specifying it in the yaml as you generate the description is nicer.

**Comment by syou6162** [comment]
> Using force inheritance is too aggressive if a downstream child has a proper description. So this is a happy middle ground I think. We could consider a flag that accepts. comma separated list of column_names to exclude from propagation which gives you intra-model control but I think specifying it in the yaml as you generate the description is nicer.

@z3z1ma Thank you very much. It would have been nice to discuss the issue beforehand, and then discuss the background and specifications. I apologize for sending the pull request suddenly and without consulting you.

Thanks also for the merge, and I look forward to the next release!

**Pull request merged**